hazelcast/src/main/java/com/hazelcast/executor/impl/ExecutorServiceProxy.java

```diff
@@ -550,17 +566,26 @@ public void shutdown() {
         Collection<Member> members = nodeEngine.getClusterService().getMembers();
         OperationService operationService = nodeEngine.getOperationService();
         Collection<Future> calls = new LinkedList<Future>();
+        QuorumException exception = null;
 
         for (Member member : members) {
             if (member.localMember()) {
-                getService().shutdownExecutor(name);
+                try {
+                    getService().ensureQuorumPresent(getName(), QuorumType.WRITE);
+                    getService().shutdownExecutor(name);
+                } catch (QuorumException ex) {
+                    exception = ex;
+                }
             } else {
                 Future f = submitShutdownOperation(operationService, member);
                 calls.add(f);
             }
         }
+        if (exception != null) {
+            throw exception;
+        }
 
-        waitWithDeadline(calls, 1, TimeUnit.SECONDS, WHILE_SHUTDOWN_EXCEPTION_HANDLER);
+        waitWithDeadline(calls, 3, TimeUnit.SECONDS, shutdownExceptionHandler);
```

Comment:
Why is the deadline increased?

Reply:
it's just not enough for a simple operation to finish on my local host, so I didn't even manage to get it to OperationRunner. 1 second was too short imho, I changed it to a value thomas used in scheduled one I think